feat: upgrade @mastra/fastembed to AI SDK v5 specification

camachoh1 opened the pull request:
Fixes #8485

Updated `@mastra/fastembed` to use AI SDK v5 (specification version v2) for compatibility with `@mastra/core`. The default exports now use v2, while legacy v1 exports are still available for backwards compatibility.

Users can now use fastembed with the latest AI SDK v5:

```typescript
import { embed } from 'ai';
import { fastembed } from '@mastra/fastembed';

const result = await embed({
  model: fastembed,
  value: 'Text to embed',
});
```

For backwards compatibility with v1, use the legacy exports:

```typescript
import { fastembed } from '@mastra/fastembed';

const legacyModel = fastembed.smallLegacy; // v1 spec
const newModel = fastembed.small; // v2 spec (default)
```

Added comprehensive integration tests to ensure both v2 and legacy v1 models work correctly.

Co-authored-by: Richard LoRicco <rtlo9962@gmail.com>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Default export now targets AI SDK v2 (AI SDK v5); added legacy model exports for v1 compatibility (smallLegacy, baseLegacy).

* **Documentation**
  * Expanded README with v2/v1 compatibility notes, available models, default (v2) usage, and direct AI SDK v5 examples.

* **Tests**
  * Added comprehensive tests validating v2 behavior, legacy v1 error paths, multi-value embeddings, and integration patterns.

* **Chores**
  * Updated SDK dependency versions, added provider tooling dependency, and added a fastembed test script.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

camachoh1 committed:
Upgraded @mastra/fastembed to use AI SDK v5 (v2 specification) for compatibility with @mastra/core. Legacy v1 exports are available for backwards compatibility.

camachoh1 committed:
refactor: reverse AI SDK dependency aliases (ai=v5, ai-v4=v4)

NikAiyer reviewed (commented):
LGTM! I think we just need to do Daniel's suggestion and this should be good to go!

abhiaiyer91 committed:
Merge branch 'main' into fix/fastembed-ai-sdk-v2-compatibility

NikAiyer reviewed (approved):
Hey @camachoh1 can you resolve the conflicts and add a changeset? After that this should be good to go!

camachoh1 committed:
chore: add changeset for fastembed AI SDK v5 upgrade

camachoh1 committed:
Merge branch 'main' of github.com:mastra-ai/mastra into fix/fastembed-ai-sdk-v2-compatibility

abhiaiyer91 committed:
Merge branch 'main' into fix/fastembed-ai-sdk-v2-compatibility

camachoh1 committed:
fix: resolve linting errors in fastembed integration tests

- Reorder imports to place @mastra/fastembed before external packages
- Prefix unused variable with underscore to indicate intentional non-use

camachoh1 committed:
test: tighten error assertion in legacy model compatibility test

Replace broad regex pattern with specific pattern that matches
"unsupported model version" error message from AI SDK v5

camachoh1 committed:
fix: remove local pnpm-lock.yaml to use monorepo lockfile

The integration-tests-v5 package had a stale local lockfile
with outdated vite references. Remove it to use the root
monorepo lockfile which is the correct approach.

camachoh1 committed:
fix: regenerate pnpm-lock.yaml to remove stale vite dependency

Run pnpm install --ignore-workspace to regenerate the lockfile

DanielSLew committed:
Merge branch 'main' into fix/fastembed-ai-sdk-v2-compatibility

NikAiyer committed:
Merge branch 'main' into fix/fastembed-ai-sdk-v2-compatibility

camachoh1 committed:
style: fix prettier formatting in fastembed package

Apply prettier formatting to README.md and index.ts

DanielSLew committed:
Merge branch 'main' into fix/fastembed-ai-sdk-v2-compatibility
